File: QA.ProductCatalog.Admin.WebApp/ReactViews/Shared/Types/TaskItem.ts
import { TaskState } from "../Enums";
import { TaskMessage } from './TaskMessage';

export default interface TaskItem {
  ChannelLanguage: string;
  ChannelState: string;
  ChannelDate: Date;
  IsDefault: boolean;
  TaskId: number;
  TaskState: TaskState;
  TaskProgress: number;
  TaskStart: string; // Date string
  TaskEnd: Date;
  TaskMessage: string | TaskMessage; // Backend returns object as a string, we parse is every time for in-app use
}

File: QA.ProductCatalog.HighloadFront/Models/TaskItem.cs
﻿using QA.Core.ProductCatalog.ActionsRunnerModel;
using System;

namespace QA.ProductCatalog.HighloadFront.Models
{
    public class TaskItem
    {
        public string ChannelLanguage { get; set; }
        public string ChannelState { get; set; }
        public DateTime? ChannelDate { get; set; }
        public bool IsDefault { get; set; }
        public int? TaskId { get; set; }
        public State? TaskState { get; set; }
        public int? TaskProgress { get; set; }
        public DateTime? TaskStart { get; set; }
        public DateTime? TaskEnd { get; set; }
        public string TaskMessage { get; set; }

    }
}